### ISSUE TITLE
bal -S won't show large revenues first, even with --invert

### ISSUE THREAD

**Issue by simonmichael** [simonmichael/hledger#1283]
Since revenues are negative, `hledger bal revenues -S` shows the smallest ones at the top. I think maybe it should reverse them, showing the largest absolute magnitudes at the top (with or without `--invert`).

**Comment by Xitian9** [comment]
I agree. I think anything fancier will be more trouble than it's worth

I also think the sorting code can be simplified given the new multiBalanceReport code. I'll look into this.

**Comment by Xitian9** [comment]
A related issue:
`hledger bal assets -S --historical` will sort alphabetically (as will `--cumulative`). This is due to the totals column being set to zero for these kinds of reports. Perhaps a more meaningful number for the total is the last column for cumulative and historical reports.

**Comment by simonmichael** [comment]
> hledger bal assets -S --historical will sort alphabetically (as will --cumulative). 

Wow, good find. I didn't reproduce that exactly, but I think I did reproduce a lack of amount-sorting even in a non-tabular balance report. We should probably open a new issue.

**Comment by Xitian9** [comment]
A minor issue. MixedAmount does not currently support `abs`. I presume `map abs` is not used because there is not a sensible definition of `signum` which would respect the law that `signum x * abs x == x`. I imagine it's fine to just define `abs` as above and leave `signum` undefined. Correct me if you feel otherwise.

**Comment by Xitian9** [comment]
Perhaps an option is, when sorting with -S, to partition each level of accounts into positive and negative amounts. Then place the positive accounts first, in decreasing order, and after that the negative accounts, in decreasing order of magnitude. Thoughts?

**Comment by simonmichael** [comment]
Example:
```
1/1
 (a)   100
 (b)  -100
 (c)   200
 (d)  -200
```
Unsorted:
```
$ hledger bal -N
                 100  a
                -100  b
                 200  c
                -200  d
```
Current amount sorting:
```
$ hledger bal -N -S
                 200  c
                 100  a
                -100  b
                -200  d
```
#1291 amount sorting
```
$ hledger bal -N -S
                 200  c
                 100  a
                -200  d
                -100  b
```
"Most positive first, followed by most negative first". It feels a bit weird. Am I just looking at it wrong ?

**Comment by simonmichael** [comment]
With more realistic account names:
```
$ hledger bal -f a.j -N 
                 100  expenses:a
                 200  expenses:c
                -100  revenues:b
                -200  revenues:d
$ hledger bal -f a.j -N -S
                 200  expenses:c
                 100  expenses:a
                -100  revenues:b
                -200  revenues:d
$ stack exec -- hledger bal -f a.j -N -S
                 200  expenses:c
                 100  expenses:a
                -200  revenues:d
                -100  revenues:b
```

**Comment by simonmichael** [comment]
And using incomestatement instead of balance. Because it normalises signs, this sorting fix isn't actually needed, but seems harmless:
```
$ hledger is -f a.j -N 
Income Statement 2020-01-01

            || 2020-01-01 
============++============
 Revenues   ||            
------------++------------
 revenues:b ||        100 
 revenues:d ||        200 
============++============
 Expenses   ||            
------------++------------
 expenses:a ||        100 
 expenses:c ||        200 
$ hledger is -f a.j -N -S
Income Statement 2020-01-01

            || 2020-01-01 
============++============
 Revenues   ||            
------------++------------
 revenues:d ||        200 
 revenues:b ||        100 
============++============
 Expenses   ||            
------------++------------
 expenses:c ||        200 
 expenses:a ||        100 
$ stack exec -- hledger is -f a.j -N -S
Income Statement 2020-01-01

            || 2020-01-01 
============++============
 Revenues   ||            
------------++------------
 revenues:d ||        200 
 revenues:b ||        100 
============++============
 Expenses   ||            
------------++------------
 expenses:c ||        200 
 expenses:a ||        100 
```

**Comment by simonmichael** [comment]
I'm inclined to think the proposed change to `bal -S` would be more of a cost than a benefit. Some alternatives:

- add a --reverse flag, that just reverses whatever -S did
- leave things alone; if anyone ever complains about bal -S, tell them to use bs/is, or unix tools

**Comment by simonmichael** [comment]
After #1379 and new doc https://hledger.org/hledger.html#sorting-by-amount, I think we can close this ?

**Comment by Xitian9** [comment]
Sounds reasonable to me.
